modules/compute/vm-instance/variables.tf

```diff
@@ -314,25 +314,30 @@ variable "placement_policy" {
   Control where your VM instances are physically located relative to each other within a zone.
   See https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_resource_policy#nested_group_placement_policy
   EOT
-
-  type    = any # It's a workaround of lack of `optional` in Terraform 1.2
+  type = object({
+    vm_count                = optional(number)
+    availability_domain_count = optional(number)
+    collocation             = optional(string)
+    max_distance            = optional(number)
+    gpu_topology            = optional(string)
+  })  
   default = null
   validation {
```

Comment:
With the new type check, the validation blocks are not needed. The type checking will ensure the placement_policy is either null or a map of provided keys with each value of the specified type.

Please ensure that this type validation is working on terraform version 1.5.7 (the ctk supported version). You can remove the valdiation blocks after  testing to ensure that wrong values are flagged by the type checker